everestjs/everestjs.cpp

```diff
@@ -93,8 +93,7 @@ static Napi::Value setup_cmd_handler(const std::string& impl_id, const std::stri
                     std::vector<napi_value> args{ctx->cmd_handlers[cmd_key].Value(), ctx->js_module_ref.Value(), arg};
                     return args;
                 },
-                [&result](const Napi::Value& retval, bool) {
-                    // FIXME (aw): we could check the bool for failure ...
+                [&result](const Napi::Value& retval) {
```

Comment:
I'm not quite sure if this is the right way to go.  Because the on_reject isn't set now anymore, we have no chance to set a custom rejection handler.  I guess now we will only receive an exception, which we can't directly identify, i.e. from which async callback it came from - this would have been possible if we registered custom logic in the on_reject function.
So why we don't check the bool here? If it is true, we know that we're in the on_reject branch and retval should be our error object, which we can handle (that's why there was the FIXME comment) or just throw it right now.

Reply:
no, to the contrary: we now have a complete js backtrace pointing us directly to the line of code that caused the exception.

Reply:
Second: why catch it and rethrow it manually when we could leave it to the nodejs to do exactly that?

Reply:
But where would/could you catch this exception now?  If you want to have the backtrace, you could have just written
if (bool == true) throw Napi::Error
or something similar.  But I would leave it up to the user to decide on how to deal with rejected promises when using the JsExecContext helper class